File: runtime/tooling/tree-sitter-phpx/bindings/node/binding_test.js
/// <reference types="node" />

const assert = require("node:assert");
const { describe, it } = require("node:test");

const Parser = require("tree-sitter");
const { phpx, phpx_only } = require("../..");

describe("PHPX", () => {
  const parser = new Parser();
  parser.setLanguage(phpx);

  it("should be named phpx", () => {
    assert.strictEqual(parser.getLanguage().name, "phpx");
  });

  it("should parse source code", () => {
    const sourceCode = "<?php echo 'Hello, World!';";
    const tree = parser.parse(sourceCode);
    assert(!tree.rootNode.hasError);
  });
});

describe("PHPX Only", () => {
  const parser = new Parser();
  parser.setLanguage(phpx_only);

  it("should be named phpx_only", () => {
    assert.strictEqual(parser.getLanguage().name, "phpx_only");
  });

  it("should parse source code", () => {
    const sourceCode = "echo 'Hello, World!';";
    const tree = parser.parse(sourceCode);
    assert(!tree.rootNode.hasError);
  });
});

File: runtime/tooling/tree-sitter-phpx/bindings/go/binding_test.go
package tree_sitter_phpx_test

import (
	"testing"

	tree_sitter "github.com/tree-sitter/go-tree-sitter"
	tree_sitter_phpx "github.com/samifouad/deka/tooling/tree-sitter-phpx/bindings/go"
)

func TestPHPGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_phpx.LanguagePHPX())
	if language == nil {
		t.Errorf("Error loading PHPX grammar")
	}

	sourceCode := []byte("<?php echo 'Hello, World!';")
	parser := tree_sitter.NewParser()
	defer parser.Close()
	parser.SetLanguage(language)

	tree := parser.Parse(sourceCode, nil)
	if tree == nil || tree.RootNode().HasError() {
		t.Errorf("Error parsing PHP")
	}
}

func TestPHPOnlyGrammar(t *testing.T) {
	language := tree_sitter.NewLanguage(tree_sitter_phpx.LanguagePHPXOnly())
	if language == nil {
		t.Errorf("Error loading PHPX-only grammar")
	}

	sourceCode := []byte("echo 'Hello, World!';")
	parser := tree_sitter.NewParser()
	defer parser.Close()
	parser.SetLanguage(language)

	tree := parser.Parse(sourceCode, nil)
	if tree == nil || tree.RootNode().HasError() {
		t.Errorf("Error parsing PHP")
	}
}
